romdisk/date.asm

```diff
@@ -13,25 +13,63 @@
         EXTERN init_static_buffer
         DEFC STATIC_DATE_BUFFER = init_static_buffer
 
-        ; date main routine. 
-        ; Print the current date and time on screen
+        ; Date main routine, print the current date and time on screen.
+        ; With zero arguments, just print date.
+        ; With one argument, set date; with two arguments, set date and time.
+        ; Date and time format is:
+        ;       YYYY-MM-DD HH:MM:SS
         ; Parameters:
         ;       HL - ARGV
         ;       BC - ARGC
         ; Returns:
         ;       A - 0 on success
         PUBLIC date_main
 date_main:
-        ; Ignore parameters
+        ; Read the existing date whether we are reading or writing
+        ld de, STATIC_DATE_BUFFER
         ; Get system date:
         ;   DE - Address of the date structure to fill. Must be at least DATE_STRUCT_SIZE bytes.
         ; Returns:
         ;   A - ERR_SUCCESS on success, error code else
-        ld de, STATIC_DATE_BUFFER
+        push hl
         GETDATE()
+        pop hl
+        or a
+        jp nz, date_error
+        dec bc ; Discount command name
+        ld a, b ; See if there are any arguments
+        or c
+        jr z, _date_main_print ; no args, print
+        ; Parse date
+        inc hl ; Skip command name
+        inc hl
+        ld e, (hl) ; Load date string into DE
+        inc hl
+        ld d, (hl)
+        inc hl
+        push hl
+        ld hl, STATIC_DATE_BUFFER
+        call ascii_to_date
+        ; We could make the parser do some checks and test here
+        pop hl
+        ; See if there is also a time
+        dec bc
+        ld a, b
+        or c
+        jr z, _do_setdate_syscall
+        ; Parse time
+        ld e, (hl) ; Load time string into  DE
+        inc hl
+        ld d, (hl)
+        ld hl, STATIC_DATE_BUFFER
```

Comment:
You should load directly the destination of time in the structure here:
```
ld hl, STATIC_DATE_BUFFER + TIME_OFFSET_IN_STRUCT
```

Where `DEFC TIME_OFFSET_IN_STRUCT = 5` is defined at the top of the file